components/RegionDialog/RegionDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import React from "react";
+import { useForm } from "react-hook-form";
+
+import { clearCheckout } from "@/lib/checkout";
+import { DEFAULT_LOCALE, LOCALES } from "@/lib/regions";
+
+import { Button } from "../Button";
+import useChannels from "../ChannelsProvider/useChannels";
+
+interface RegionDialogProps {
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+export interface RegionFormData {
+  channel: string;
+  locale: string;
+}
+
+const RegionDialog = ({ isOpen, onClose }: RegionDialogProps) => {
+  const router = useRouter();
+  const { channels, currentChannel, setCurrentChannel } = useChannels();
+
+  const { register, handleSubmit } = useForm<RegionFormData>({
+    defaultValues: {
+      channel: currentChannel.slug,
+      locale: router.locale || DEFAULT_LOCALE,
```

Comment:
should `router.locale` ever be null or undefined?

Reply:
When user navigates to the rootpath without the locale/channel